src/cli/fillerconvert/structures/state_test_filler.py

```diff
@@ -0,0 +1,155 @@
+
+            # Convert pre state data from .json filler into pyspec type
+            pre = Alloc()
+            for account_address, account in state_test.pre.items():
+                storage: Storage = Storage()
+                for key, value in account.storage.items():
+                    storage[key] = value
+
+                pre[account_address] = Account(
+                    balance=account.balance,
+                    nonce=account.nonce,
+                    code=Bytes.fromhex(account.code[2:]),
+                    storage=storage,
+                )
+
+            # for each transaction compute test vector
+            for d, _ in enumerate(state_test.transaction.data, start=0):
+                for g, _ in enumerate(state_test.transaction.gas_limit, start=0):
+                    for v, _ in enumerate(state_test.transaction.value, start=0):
+                        for expect in state_test.expect:
+                            if expect.has_index(d, g, v):
+                                for fork in expect.network:
+                                    vector = self._make_vector(
+                                        test_name,
+                                        state_test,
+                                        fork,
+                                        env,
+                                        pre,
+                                        d,
+                                        g,
+                                        v,
+                                        expect.result,
+                                    )
+                                    vectors.append(vector)
+        return vectors
+
+    def _make_vector(
+        self,
+        test_name,
+        state_test: StateTestInFiller,
+        fork,
+        env,
+        pre,
+        d: int,
+        g: int,
+        v: int,
+        expect_result: Dict[AddressInFiller, AccountInExpectSection],
+    ) -> StateTestVector:
+        """Compose test vector from test data."""
+        vector = StateTestVector()
+        vector.env = env
+        vector.pre = pre
+
+        tr: Transaction = Transaction()
+        data = state_test.transaction.data[d]
+        tr.data = Bytes.fromhex(data[2:])
+        tr.gas_limit = HexNumber(state_test.transaction.gas_limit[g])
+        tr.value = HexNumber(state_test.transaction.value[v])
+        tr.gas_price = HexNumber(state_test.transaction.gas_price)
+        tr.nonce = HexNumber(state_test.transaction.nonce)
+        tr.to = Address(state_test.transaction.to)
+        tr.secret_key = Hash(state_test.transaction.secret_key)
+        vector.tx = tr
+
+        post = Alloc()
+        for address, account in expect_result.items():
+            storage = Storage()
+            if account.storage is not None:
+                for key, value in account.storage.items():
+                    storage[key] = value
+
+            # TODO looks like pyspec post state verification will not work for default values?
```

Comment:
We can use `model_fields_set` of the `account` to see if a value was explicitly set.

For example, if `"balance" in account.model_fields_set`, it means the value was explicitly set to zero instead the value being zero because if fell back to the default.